pkg/node/node.go

```diff
@@ -0,0 +1,182 @@
+		ctx:       ctx,
+		clientset: clientset,
+		log:       logger,
+	}
+}
+
+func (n *Node) Update(csi *csibaremetalv1.Deployment, scheme *runtime.Scheme) error {
+	var (
+		// need to trying deploy each daemonset
+		// return err != nil to request reconcile again if one ore more daemonsets failed
+		resultErr error
+		namespace = common.GetNamespace(csi)
+	)
+
+	needToDeploy, err := n.updateNodeLabels()
+	if err != nil {
+		return err
+	}
+
+	for platformName, isDeploying := range needToDeploy {
+		if isDeploying {
+			expected := createNodeDaemonSet(csi, platforms[platformName])
+			if err := controllerutil.SetControllerReference(csi, expected, scheme); err != nil {
+				n.log.Error(err, "Failed to set controller reference "+expected.Name)
+				continue
+			}
+
+			if err = n.updateDaemonset(expected, namespace); err != nil {
+				n.log.Error(err, "Failed to update daemonset "+expected.Name)
+				resultErr = err
+			}
+		}
+	}
+
+	return resultErr
+}
+
+// CleanLabels deletes platform-label on each node in cluster
+func (n *Node) CleanLabels() error {
+	return n.cleanNodeLabels()
+}
+
+func (n *Node) updateDaemonset(expected *v1.DaemonSet, namespace string) error {
+	dsClient := n.clientset.AppsV1().DaemonSets(namespace)
+
+	found, err := dsClient.Get(n.ctx, expected.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if _, err := dsClient.Create(n.ctx, expected, metav1.CreateOptions{}); err != nil {
+				n.log.Error(err, "Failed to create daemonset "+expected.Name)
+				return err
+			}
+
+			n.log.Info("Daemonset created successfully: " + expected.Name)
+			return nil
+		}
+
+		n.log.Error(err, "Failed to get daemonset "+expected.Name)
+		return err
+	}
+
+	if common.DaemonsetChanged(expected, found) {
+		found.Spec = expected.Spec
+		if _, err := dsClient.Update(n.ctx, found, metav1.UpdateOptions{}); err != nil {
+			n.log.Error(err, "Failed to update daemonset "+expected.Name)
+			return err
+		}
+
+		n.log.Info("Daemonset updated successfully: " + expected.Name)
+		return nil
+	}
+
+	return nil
+}
+
+// updateNodeLabels gets list of all nodes in cluster,
+// selects fit platform for each one and add/update node platform-label
+// returns a Set of platforms, which will be deployed
+func (n *Node) updateNodeLabels() (Set, error) {
+	// need to trying getKernelVersion and update label on each node
+	// return err != nil to request reconcile again if one ore more nodes failed
+	var resultErr error
+
+	needToDeploy := createPlatformsSet()
+
+	nodes, err := n.clientset.CoreV1().Nodes().List(n.ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, node := range nodes.Items {
+		kernelVersion, err := GetNodeKernelVersion(node)
+		if err != nil {
+			n.log.Error(err, "Failed to get kernel version for "+node.Name)
+			resultErr = err
+			continue
+		}
+
+		platformName := findPlatform(kernelVersion)
+		needToDeploy[platformName] = true
+
+		node.Labels[label] = platforms[platformName].labeltag
```

Comment:
Do we need to remove this logic for node controller?

Reply:
Yes, we do